code/modules/cm_marines/m2c.dm

```diff
@@ -589,6 +606,95 @@
 		rotate_to(user, target)
 		return TRUE
 
+/obj/structure/machinery/m56d_hmg/auto/t37
+	name = "\improper T37 Medium Machinegun"
+	desc = "A deployable, medium machine gun. The T37 is a UPP machinegun that fires 7.62x64mmR rounds. Unlike the USCM M2C counterpart, it has a much slower firerate in favor of sustained fire. Due to the quality of the design, it has a tendency to explode if it sustains too much damage. <B> Click its sprite while behind it without holding anything to man it. Click-drag on NON-GRAB intent to disassemble the gun, GRAB INTENT to remove ammo magazines."
+	icon_state = "t37"
+	icon_full = "t37"
+	icon_empty = "t37_e"
+	rounds_max = 150
+	ammo = /datum/ammo/bullet/machinegun/auto/medium
+	fire_delay = 0.35 SECONDS
+	grip_dir = null
+	stationary = TRUE
+
+	// OVERHEAT MECHANIC VARIABLES
+	overheat_threshold = 20
+	passive_cooldown = T37_PASSIVE_COOLDOWN_AMOUNT
+
+	var/explosion_chance = 70
+	var/explosion_power = 65
+	var/explosion_falloff = 25
+	var/falloff_mode = EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL
+
+/obj/structure/machinery/m56d_hmg/auto/t37/update_health(amount)
+	health -= amount
+	if(health <= 0)
+		if(prob(explosion_chance))
+			var/cause_data = create_cause_data(src.name)
+			visible_message(SPAN_HIGHDANGER("[src] explodes into pieces!"))
+			cell_explosion(loc, explosion_power, explosion_falloff, falloff_mode, null, cause_data)
+		playsound(src.loc, 'sound/items/Welder2.ogg', 25, 1)
+		qdel(src)
+		return
+	else
+		..()
+
+/obj/structure/machinery/m56d_hmg/auto/t37/handle_ammo_out(mob/user)
+	visible_message(SPAN_NOTICE("[icon2html(src, viewers(src))] [src]'s ammo box drops onto the ground, now completely empty."))
+	playsound(loc, empty_alarm, 70, 1)
+	update_icon() //final safeguard.
+	var/obj/item/ammo_magazine/m2c/t37/AM = new /obj/item/ammo_magazine/m2c/t37(src.loc)
+	AM.current_rounds = 0
+	AM.update_icon()
+
+/obj/structure/machinery/m56d_hmg/auto/t37/attackby(obj/item/O as obj, mob/user as mob)
+	if(!ishuman(user))
+		return
+	// RELOADING
+	if(istype(O, /obj/item/ammo_magazine/m2c/t37))
```

Comment:
are we overriding this whole proc just to change this line? ideally we would change the parent proc in /obj/structure/machinery/m56d_hmg/auto like:

```dm

if(istype(O, /obj/item/ammo_magazine)
	var/obj/item/ammo_magazine/magazine = O
	
	if(magazine.default_ammo != ammo)
		return
```